### PR TITLE
MD5 File checksums

### PR DESCRIPTION
**Pull request by pysysops**
Hey @aelsabbahy,

I thought a good addition to the `file` resource would be the ability to validate an md5 checksum of the file. It was pretty simple to do. 

I've added the appropriate code, docs and a test file / goss test. 

Output on error looks like: 
```
Failures/Skipped:

File: /etc/passwd: md5:
Expected
    <string>: 6b0364b78fb8968a2d15879125556c3c
to equal
    <string>: 12341234

Total Duration: 1.012s
Count: 52, Failed: 1, Skipped: 0'
```

Errors such as validating an md5 on a directory are handled:
```
Failures/Skipped:

/tmp: md5: Error: read /tmp: is a directory

Total Duration: 1.013s
Count: 54, Failed: 1, Skipped: 0'
```

Any feedback appreciated. Thanks!

### CODE REVIEW

**Comment by aelsabbahy** [comment]
Thanks for the awesome contributions so far. I'll try to process all three outstanding PRs early next week.

**Comment by pysysops** [comment]
Thanks for a really cool project. It's got great potential and it's simplicity / ease of use is a big WIN.

As I think of useful enhancements I'll either open PR's or discuss them as issues as well as try to answer any user issues that I can.

File: system/file.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ import (
 	"strconv"
```
Comment (aelsabbahy) [inline system/file.go]: Please run this through `goimports` to fix indentation and ordering of imports.

File: system/file.go [hunk 2]
```diff
@@ -214,3 +216,23 @@ func realPath(path string) (string, error) {
 
 	return realPath, err
 }
+
+func (f *DefFile) Md5() (string, error) {
```
Comment (aelsabbahy) [inline system/file.go]: The `result` variable can be dropped altogether, this code should do the same as the orig:
```golang
func (f *DefFile) Md5() (string, error) {
        if err := f.setup(); err != nil {
                return "", err
        }

        fh, err := os.Open(f.realPath)
        if err != nil {
                return "", err
        }
        defer fh.Close()

        hash := md5.New()
        if _, err := io.Copy(hash, fh); err != nil {
                return "", err
        }

        return fmt.Sprintf("%x", hash.Sum(nil)), nil
}
```

Comment (pysysops) [inline system/file.go]: Ah yes... That's clear after another look at it 👍 this was a super quick addition initially.

**Pull request merged**